### DISCUSSION TITLE
Error in the whisper model.transcribe function

### DISCUSSION THREAD

**Discussion by actlceo** [openai/whisper#2123] [Q&A]
I am rusty at coding novice at python workign with co-pilot ChatGPT

I am running the following code part of my python code
 model = whisper.load_model("base")

    # Transcribe the audio
    print(f"sucess:  moved into transcribe function {audio_path}")
   # result = model.transcribe(audio_path)
    result = model.transcribe(str(audio_path))  # Convert path to string

the  debug messages show that the file is there:
(sucess: File found at C:\Users\saurabh\Desktop\python\EarningsCall.wav
sucess:  moved into transcribe function C:\Users\saurabh\Desktop\python\EarningsCall.wav)
but then I get the error in the model.transcribe fucntion
 file not found 
  File "C:\Program Files\Python312\Lib\subprocess.py", line 1538, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [WinError 2] The system cannot find the file specified

the wav file is fine. I actually am using this from the OpenAI api site:-) 
asked OpenAi chat it says everything looks fine, seems to be some sort of issue with whisper so I should post it here :-) 

can anyone please help . thanks

**Comment by danka12324** [comment]
Your problem is related to the fact that the Whisper model's transcribe function expects the path to the file, not the file itself. When you pass str(audio_path), you convert the path to a string, but do not provide the file itself. You need to pass the file name and then open it for reading.

Try the following code:


import whisper

# Replace "base" with the version of the model you want to use
model = whisper.load_model("base")

# Transcribing an audio
file print(f"Success: moving to the transcription function {audio_path}")
with open(audio_path, 'rb') as audio_file:
result = model.transcribe(audio_file)

# It is assumed that the result has already been processed and output
In this code, we open the file for reading in binary mode ('rb') and pass the file object to the transcribe function. This will allow the Whisper model to read the data from the file and transcribe it.

**Reply by achaosss** [reply]
I user the code, but got error.
My package version is openai-whisper==20231117

expected np.ndarray (got _io.BufferedReader)
  File "C:\Users\Z19070948\Desktop\test\whisper_test.py", line 6, in <module>
    result = model.transcribe(audio_file)
TypeError: expected np.ndarray (got _io.BufferedReader)
